tools/autograd/gen_variable_type.py

```diff
@@ -304,6 +304,11 @@
 }
 """)
 
+FW_DERIVATIVE_BOOLEAN_TEMPLATE = CodeTemplate("""\
+auto ${boolean_var_name} = ${cond};
+(void)${boolean_var_name};
+""")
```

Comment:
Is there a way to check that the compiler optimized away this code if it's not used?
```
auto _any_has_forward_grad_result = isFwGradDefined(self) || isFwGradDefined(other);
(void)_any_has_forward_grad_result;
```
I remember sometimes the compiler gets confused about if it's safe to optimize away things (e.g. if the compiler thinks isFwGradDefined has any side effects)

Reply:
From checking the assembly code, the calls to isFwGradDefined are not optimized out indeed.
I'll add the appropriate logic to avoid these calls for out= functions

Reply:
RIP, I wish the compiler were smarter

Reply:
I think it could be if they were inlined and only calling simple functions, but that's not the case here :/